src/Jobs/AbstractBatchedActionSchedulerJob.php

```diff
@@ -67,15 +67,18 @@ public function handle_create_batch_action( int $batch_number ) {
 		$this->monitor->validate_failure_rate( $this );
 
 		$items = $this->get_batch( $batch_number );
+		$count = $this->unfiltered_count ?? count( $items );
```

Comment:
I don't think it's a good idea to add `unfiltered_count` here. This class is supposed to be abstract and we're now including child class' logic into it.

What we can do instead is to override the `handle_create_batch_action` in classes that need them and, for example, instead of having a global `unfiltered_count` variable, we pass it as an argument. We might also be able to use a trait to define a common method and reuse that one.

I'm just not sure if WordPress' `add_filter` function called in the `init` method will pick up the child's `handle_create_batch_action` method instead of the parent's. But even if it doesn't we could override the `init` method in child class.

Reply:
Thanks, it makes sense to have the logic in the child class only. At the moment we only have the UpdateAllProducts job which uses it so no need to create a trait yet.

All these functions are defined in an interface so we can't just change the parameters, but I've added a separate function which gets a filtered batch.